### PR TITLE
Fix draging item destroy content inside tooltip

### LINKED ISSUE #12532: Tooltip needs to be hidden before sorting menu order
### Steps to Reproduce

1. open the bakery demo admin panel
2. click on bread page
3. actions -> sort menu order
4. open the tool tip (context menu with 3 dots) of e.g. bagel page
5. with the tool tip still open, drag and drop the bagel page to change its order
6. the item occupies much more space than it should be, including the tool tip (unfortunately the video doesn't show that). if you try to open the tool tip again, it will only show a small triangle at the bottom

https://github.com/user-attachments/assets/baf29123-a21b-47b1-b740-dc5adf9e7042

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.12.7
-   Django version: 5.0.9
-   Wagtail version: 6.4a0
-   Browser version: https://www.whatsmybrowser.org/b/ERLNT (chrome 128 on arch linux)

### Working on this

I've tried to fix it on my own  but haven't been successful so far. I've tried adding a plugin on https://github.com/wagtail/wagtail/blob/6f34d3c29977ede3820603a7db444871cd6fa849/client/src/controllers/DropdownController.ts#L287-L292 and dispatching an event from the `OrderableController` but for some reason it never dispatches that event. In my opinion it is basically the same as `hideTooltipOnBreadcrumbsChange`. If I could just find out why it doesn't dispatch my events.

### PR DESCRIPTION
**Pull request by ayaan-qadri**
_Please describe the problem you're fixing here. Include the issue number, if applicable._

### Fixes #12532 

Issue break down:

- In the stimulus controller lifecycle, the `disconnect` method runs when an item is dragged (in our case) ([1st point in the doc ref](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#disconnection)), and then the `connect` method.  
- The `disconnect` method calls the `destroy` method of tippy, which completely removes the tippy instance which has the tooltip content ([doc ref](https://atomiks.github.io/tippyjs/v5/methods/#destroy)).  
- After the `disconnect` method is executed, the `connect` method is called again (as the item is reattached after being dragged) and `connect` method creates new tippy instance.  
- Issue occurs here
    - However, since the previous  tippy instance with tooltip content was completely destroyed by `destroy` method, the newly created tippy instance does not include tooltip content or completely new tippy instance with no tooltip content.

Solution that implemented:
- Hiding tippy instead destroy so that we can reuse it and condition check if tippy instance exists then do not create new one.

<details>
<summary>(Video from issue author)</summary>

https://private-user-images.githubusercontent.com/41559987/382435150-baf29123-a21b-47b1-b740-dc5adf9e7042.mp4

</details>

<details>
<summary>Works fine now</summary>

https://github.com/user-attachments/assets/d2832c83-6949-48b4-967e-ccc76c4ed477

</details>

### CODE REVIEW

File: client/src/controllers/DropdownController.ts [hunk 1]
```diff
@@ -307,6 +309,6 @@ export class DropdownController extends Controller<HTMLElement> {
   }
 
   disconnect() {
-    this.tippy?.destroy();
+    this.tippy?.hide();
```
Comment (laymonage) [inline client/src/controllers/DropdownController.ts:312]: I'm not sure this is the right solution. Keeping the Tippy instance indefinitely after the Stimulus controller is disconnected can cause memory leak issues. We have places where the controller is replaced multiple times, e.g. in the concurrent editing notifications.

Maybe the solution is to somehow make sure that the necessary options/elements to instantiate Tippy is retained after the initial render and after the drag-and-drop. Not sure if it does, but the `keepMounted` value may help.

Comment (laymonage) [reply client/src/controllers/DropdownController.ts:312]: Some features like the concurrent editing notifications (see `SessionController`) works by swapping the HTML with a freshly-rendered fragment from the server. If the fragment contains an element with `data-controller="w-dropdown"`, I doubt `this` will be the same object as the one from the previous fragment. For example, in the `SessionController`, a dropdown might only be used if there are sessions that have made changes, or if there are many other active sessions. When a swap occurs, it may or may not even contain a dropdown.

In short, there are cases where it may not be the same controller instance that gets unmounted and remounted. If the element is unmounted and removed from the DOM, or just unmounted indefinitely, the tippy instance will only be destroyed by the garbage collector. Not to mention that we'd assume it can be garbage collected (no other leftover references to the tippy), which I can't be sure without doing some profiling.

Comment (ayaan-qadri) [reply client/src/controllers/DropdownController.ts:312]: Thank you for explaining !

Comment (ayaan-qadri) [reply client/src/controllers/DropdownController.ts:312]: @laymonage `keepMouted` is working but there is obvious issue with this, like if someone had to go through big list to change item's order so when item is dragged and on every item controller connects and disconnects which appends tippy element,

E.g,

https://github.com/user-attachments/assets/dada732e-b0d1-431f-bba2-f148a52b0447

I'm thinking for two solutions for this based on use case

1. if it is not necessary to append tippy element when `keepMounted=true` then instead append we can replace new target.
2. Or if it's necessary append instead then we can create new value data attribute (keep-mounted-is-append; default : true)  if false then not to append but replace.

https://github.com/wagtail/wagtail/blob/61b50d3c28fb55f8c56bae78bd6f7c44e866e022/client/src/controllers/DropdownController.ts#L183

**Comment by lb-** [comment]
Please also remember to include a unit test or two to ensure we cover this behaviour for any future regressions. I know it's a bit tricky to test this library but we should try.

File: client/src/controllers/DropdownController.ts (hunk 1 above)
Comment (lb-) [reply client/src/controllers/DropdownController.ts:312]: @laymonage - could you look at the response above, and help with any further direction here?

**Comment by laymonage** [comment]
This might've been fixed by the work in #12790 – I'll check again later.

**Comment by ayaan-qadri** [comment]
It is fixed but this comment still applies https://github.com/wagtail/wagtail/pull/12727#discussion_r1897460285

**Comment by ayaan-qadri** [comment]
I have a fresh setup of bakerydemo and the latest Wagtail. It seems that both the original issue for this PR and this: https://github.com/wagtail/wagtail/pull/12727#discussion_r1897460285 are already fixed in the latest version. have a look:

https://github.com/user-attachments/assets/1e68031c-cedb-414d-aacb-610b3a4bc259
